### PR TITLE
Add object type hint for docs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: scene/2d/ray_cast_2d.cpp [hunk 1]
```diff
@@ -267,10 +267,10 @@ void RayCast2D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_collision_normal"), &RayCast2D::get_collision_normal);
 
 	ClassDB::bind_method(D_METHOD("add_exception_rid", "rid"), &RayCast2D::add_exception_rid);
-	ClassDB::bind_method(D_METHOD("add_exception", "node"), &RayCast2D::add_exception);
+	ClassDB::bind_method(D_METHOD("add_exception", "node:CollisionObject2D"), &RayCast2D::add_exception);
```
Comment (neikeq) [inline scene/2d/ray_cast_2d.cpp]: Not sure if those would be ok. The method clearly excepts a CollisionObject2D, but allows you to pass any type of Object.

File: scene/gui/tree.cpp [hunk 2]
```diff
@@ -710,7 +710,7 @@ void TreeItem::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_next_visible:TreeItem"), &TreeItem::get_next_visible);
 	ClassDB::bind_method(D_METHOD("get_prev_visible:TreeItem"), &TreeItem::get_prev_visible);
 
-	ClassDB::bind_method(D_METHOD("remove_child:TreeItem", "child"), &TreeItem::_remove_child);
+	ClassDB::bind_method(D_METHOD("remove_child", "child:TreeItem"), &TreeItem::_remove_child);
```
Comment (neikeq) [inline scene/gui/tree.cpp]: Same here, it excepts a TreeItem, but allows Object.

**Review by neikeq — APPROVED** [review]

**Comment by Noshyaar** [comment]
From the reviews above, I'm not really sure. IMHO, the methods should be changed to expect the specific types if possible, unless there's an advantage I haven't realized yet.

**Comment by neikeq** [comment]
I am fine with changes like these as long as they don't break anything in GodotSharp or GDNative. Keep in mind the ClassDB API is also used to generating language bindings.
That's not the case here, but I just wonder if there is any reason in particular for the method to allow passing Object* as parameter.

**Comment by akien-mga** [comment]
I guess it's best to keep the hints as Object for methods which expect Object arguments then for now, to avoid mismatches between the actual C++ API and the bindings.

**Comment by Noshyaar** [comment]
Thanks. Fixed. :)

**Pull request merged**